python/dgl/backend/pytorch/sparse.py

```diff
@@ -970,7 +970,7 @@ class SEGMENTMM(th.autograd.Function):
     def forward(ctx, A, B, seglen_A):
         if B.dim() != 3:
             raise ValueError("segment_mm expects B to be a 3D tensor.")
-        C = th.zeros((A.shape[0], B.shape[2]), device=A.device, dtype=A.dtype)
+        C = th.empty((A.shape[0], B.shape[2]), device=A.device, dtype=A.dtype)
```

Comment:
shall we fix for `GATHERMM` too?

Reply:
The GatherMM kernel updates the output matrix C in an in-place manner. So it requires the output buffer being initialized as zeros.
https://github.com/dmlc/dgl/blob/097fd6dbb1a2ec306438edd1b94a29ec393fa0fa/src/array/cuda/gather_mm.cu#L124-L128